bootstrap_cfn/fabfile.py

```diff
@@ -0,0 +1,387 @@
+        if '{0}/32'.format(prv_ip) not in existing_hosts:
+            ec2.add_minion_to_sg(salt_sg, prv_ip)
+    groups = master_instance.get_attribute('groupSet').get('groupSet')
+    groups.append(salt_sg)
+    master_instance.modify_attribute('groupSet', [x.id for x in groups])
+
+
+@task
+def rsync():
+    if env.aws is None:
+        print "\n[ERROR] Please specify an AWS account, e.g 'aws:dev'"
+        sys.exit(1)
+    if env.environment is None:
+        print "\n[ERROR] Please specify an environment, e.g 'environment:dev'"
+        sys.exit(1)
+    if env.application is None:
+        print "\n[ERROR] Please specify an application, e.g 'application:peoplefinder'"
+        sys.exit(1)
+    if env.config is None:
+        # check if there is a deploy repo in a predefined location
+        app_yaml = '../{0}-deploy/{0}.yaml'.format(env.application)
+        if os.path.exists(app_yaml):
+            env.config = app_yaml
+        else:
+            print "\n[ERROR] Please specify a config file, e.g 'config:/tmp/sample-application.yaml'"
+            sys.exit(1)
+
+    stack_name = get_stack_name()
+    work_dir = os.path.join('..', '{0}-deploy'.format(env.application))
+    # LOAD AWS CONFIG FROM ~/.config.yaml
+    aws_config = AWSConfig(env.aws)
+
+    project_config = ProjectConfig(env.config, env.environment)
+    cfg = project_config.config
+
+    local_salt_dir = os.path.join(
+        work_dir,
+        cfg['salt'].get(
+            'local_salt_dir',
+            'salt'),
+        '.')
+    local_pillar_dir = os.path.join(
+        work_dir,
+        cfg['salt'].get(
+            'local_pillar_dir',
+            'pillar'),
+        '.')
+    local_vendor_dir = os.path.join(
+        work_dir,
+        cfg['salt'].get(
+            'local_vendor_dir',
+            'vendor'),
+        '.')
+
+    remote_state_dir = cfg['salt'].get('remote_state_dir', '/srv/salt')
+    remote_pillar_dir = cfg['salt'].get('remote_pillar_dir', '/srv/pillar')
+
+    # if not os.path.exists(local_state_dir):
+    #    shake(work_dir)
+
+    master_ip = find_master()
+    env.host_string = '{0}@{1}'.format(env.user, master_ip)
+    sudo('mkdir -p {0}'.format(remote_state_dir))
+    sudo('mkdir -p {0}'.format(remote_pillar_dir))
+    upload_project(
+        remote_dir=remote_state_dir,
+        local_dir=os.path.join(
+            local_vendor_dir,
+            '_root',
+            '.'),
+        use_sudo=True)
+    upload_project(
+        remote_dir='/srv/',
+        local_dir=os.path.join(
+            local_vendor_dir,
+            'formula-repos'),
+        use_sudo=True)
+    upload_project(
+        remote_dir=remote_state_dir,
+        local_dir=local_salt_dir,
+        use_sudo=True)
+    upload_project(
+        remote_dir=remote_pillar_dir,
+        local_dir=os.path.join(
+            local_pillar_dir,
+            env.environment,
+            '.'),
+        use_sudo=True)
```

Comment:
I think we do need a symlink here but we don't want to commit the whole file again :-)